spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/jms/activemq/ActiveMQAutoConfigurationTests.java

```diff
@@ -62,6 +62,17 @@ public void configurationBacksOffWhenCustomConnectionFactoryExists() {
 	}
 
 	@Test
+	public void customActiveMQConnectionFactoryConfiguration() {
+		load(EmptyConfiguration.class,
+				"spring.activemq.properties.nonBlockingRedelivery:true",
+				"spring.activemq.properties.producer-window-size:10");
+		ActiveMQConnectionFactory connectionFactory = this.context
```

Comment:
So that test is interesting. It looks like that ActiveMQ exposes configuration via both an explicit setter (this code) and a `properties` object (your original report). 

Now that I understand this, I am not sure really what the best course of action is. Adding properties to `ActiveMQProperties` is probably what we should do.